src/pluginscript_callbacks.lua

```diff
@@ -114,14 +123,28 @@ pluginscript_callbacks.script_init = wrap_callback(function(manifest, path, sour
 		return
 	end
 	lps_coroutine_pool:release(co)
+
 	if type(script) ~= 'table' then
 		api.godot_print_error('Script must return a table', path, path, -1)
 		return
 	end
 
+    if is_yue then
+        if script.__base ~= nil then
+            script = script.__base
+        end
+    end
+
 	local known_properties = {}
 	for k, v in pairs(script) do
-		if k == 'class_name' then
+        if is_yue and (
+                k == '__class' or
```

Comment:
Added a simple ignore for the values of `__class`, `__index` and `__base`. Let me know if there is a better way to go about this.

Reply:
Hmm, maybe ignoring anything that starts with `__`, even for Lua, makes sense? They are usually metatable stuff and is possibly not important for Godot. We'd need to warn users about this behavior, of course, but seems ok to me.